Source/ablastr/fields/PoissonSolver.H

```diff
@@ -306,66 +384,30 @@ computePhi (amrex::Vector<amrex::MultiFab*> const & rho,
         mlmg.solve( {phi[lev]}, {rho[lev]},
                      relative_tolerance, absolute_tolerance );
 
+        const amrex::IntVect& refratio = rel_ref_ratio.value()[lev];
+        const int ncomp = linop->getNComp();
+
         // needed for solving the levels by levels:
         // - coarser level is initial guess for finer level
         // - coarser level provides boundary values for finer level patch
         // Interpolation from phi[lev] to phi[lev+1]
         // (This provides both the boundary conditions and initial guess for phi[lev+1])
-        if (lev < finest_level) {
-
-            // Allocate phi_cp for lev+1
-            amrex::BoxArray ba = phi[lev+1]->boxArray();
-            const amrex::IntVect& refratio = rel_ref_ratio.value()[lev];
-            ba.coarsen(refratio);
-            const int ncomp = linop->getNComp();
-            const int ng = (grid_type == utils::enums::GridType::Collocated) ? 1 : 0;
-            amrex::MultiFab phi_cp(ba, phi[lev+1]->DistributionMap(), ncomp, ng);
-            if (ng > 0) {
-                // Set all values outside the domain to zero
-                phi_cp.setDomainBndry(0.0_rt, geom[lev]);
-            }
-
-            // Copy from phi[lev] to phi_cp (in parallel)
-            const amrex::Periodicity& crse_period = geom[lev].periodicity();
-
-            ablastr::utils::communication::ParallelCopy(
-                phi_cp,
-                *phi[lev],
-                0,
-                0,
-                1,
-                amrex::IntVect(0),
-                amrex::IntVect(ng),
-                do_single_precision_comms,
-                crse_period
-            );
-
-            // Local interpolation from phi_cp to phi[lev+1]
-#ifdef AMREX_USE_OMP
-#pragma omp parallel if (amrex::Gpu::notInLaunchRegion())
-#endif
-            for (amrex::MFIter mfi(*phi[lev + 1], amrex::TilingIfNotGPU()); mfi.isValid(); ++mfi) {
-                amrex::Array4<amrex::Real> const phi_fp_arr = phi[lev + 1]->array(mfi);
-                amrex::Array4<amrex::Real const> const phi_cp_arr = phi_cp.array(mfi);
-
-                details::PoissonInterpCPtoFP const interp(phi_fp_arr, phi_cp_arr, refratio);
-
-                amrex::Box const& b = mfi.growntilebox(ng);
-                amrex::ParallelFor(b, interp);
-            }
-
-        }
+        if (lev < finest_level) { interpolatePhiBetweenLevels(phi[lev],
```

Comment:
For cosmetic reasons, should we put `interpolatePhiBetweenLevels` on the next line and indent the arguments accordingly?